### PR TITLE
doc: cleanup/update glossary terms

### PR DESCRIPTION
**Pull request by dbkinder**
There are many graphics-related terms in the glossary that are not
referenced in the documentation.  Remove those terms, and clarify new
terms in our updated GSG and introductory docs.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -37,56 +36,37 @@ Glossary of Terms
...
+      build your application on a separate system from where the application is
+      deployed and run (i.e., the :term:`Target`). ACRN recommends using Ubuntu
+      18.04 as the OS on your development computer and that is an assumption in
+      our documentation.
+
+   Guest
+   Guest VM
+      A term used to refer to any :term:`VM` that runs on the hypervisor.  Both Service
+      and User VMs are considered Guest VMs from the hypervisor’s perspective,
+      albeit with different properties. *(You’ll find the term Guest used in the
```
Comment (fitchbe) [inline doc/glossary.rst:60]: There are a bunch of curly apostrophes (U+2019) in this file.

File: doc/glossary.rst [hunk 2]
```diff
@@ -100,98 +80,91 @@ Glossary of Terms
...
-      Operating System
-
-   OSPM
-      Operating System Power Management
-
    Passthrough Device
-      Physical devices (typically PCI) exclusively assigned to a guest.  In
-      the Project ACRN architecture, passthrough devices are owned by the
-      foreground OS.
+      Physical I/O devices (typically PCI) exclusively assigned to a User VM so
```
Comment (fitchbe) [inline doc/glossary.rst:84]: so --> so that

File: doc/glossary.rst [hunk 3]
```diff
@@ -100,98 +80,91 @@ Glossary of Terms
...
-   PV
-      Para-virtualization (See
-      https://en.wikipedia.org/wiki/Paravirtualization)
-
-   PVINFO
-      Para-Virtualization Information Page, a MMIO range used to
-      implement para-virtualization
+      A User VM configured and launched by the Service VM and typically
+      accessing shared hardware resources managed by the Service VM and Device
+      Model. Most User VMs are post-launched while special purpose User VMs are
```
Comment (fitchbe) [inline doc/glossary.rst:107]: "special purpose User VMs" --> "special-purpose User VMs"

**Review by fitchbe — APPROVED** [review]

File: doc/glossary.rst [hunk 4]
```diff
@@ -100,98 +80,91 @@ Glossary of Terms
...
+      device-access overhead issues. Such technologies include: using a
+      passthrough interrupt controller, polling-mode Virtio, Intel RDT
+      allocation features (CAT, MBA), and I/O prioritization.  RTVMs are
+      typically a :term:`Pre-launched VM`.  A non-:term:`Safety VM` with
+      real-time requirements is a :term:`Post-launched VM`.
 
    Safety VM
       A special VM with dedicated hardware resources, running in
       partition mode, and providing overall system health-monitoring
-      functionality.  Currently, a Safety VM is always a pre-launched User VM.
```
Comment (amyreye) [inline doc/glossary.rst:176]: Consider removing "running in partition mode" since "What Is ACRN" page shows Safety VM can exist in a hybrid scenario too.

**Comment by dbkinder** [comment]
Thanks @fitchbe and @amyreye.  Comments addressed and PR updated.

**Pull request merged**